R/transformCounts.R

```diff
@@ -308,6 +347,145 @@ setMethod("transformAssay", signature = c(x = "SummarizedExperiment"),
+    # estimation is not reliable with so few features.
+    if( any(found_features <= 1) ){
+        stop(
+            "There are samples that contain 1 or less found features.",
+            "'percentile' cannot be estimated from the data. Specify it ",
+            "manually.", call. = FALSE)
+    }
+    # Calculate quantiles for the features present in each sample.
+    # Use a 'probs' sequence from 0 to 1, with increments based on the maximum
+    # number of features found.
+    # This allows quantile calculation for every feature increment.
+    # The quantiles indicate how counts accumulate in each sample, showing,
+    # e.g., whether most counts are small or large.
+    quantiles <- colQuantiles(
+        mat, probs = seq(0, 1, length.out = max(found_features)), na.rm = TRUE)
+    quantiles <- t(quantiles)
+    # Sort the columns based on abundance. This means that each sample is sorted
+    # so that smallest values come first. Ultimately, this means that values
+    # should increase from first to row to last. Rows do not represent single
+    # taxa anymore but abundance level.
+    mat <- apply(mat, 2, function(col){
+        col[ is.na(col) ] <- 0
+        col <- sort(col)
+        return(col)
+    })
+    # Compute the row means of sorted matrix to create a reference profile. This
+    # profile illustrates the average abundance levels in this dataset; what
+    # abundance levels we can find from a sample in average.
+    ref <- rowMeans2(mat)
+    ref <- ref[ ref > 0 ]
+    # Calculate the differences between the reference profile and each
+    # column's quantiles. This means that we compare this average abundance
+    # levels to each samples abundance levels.
+    difference <- ref - quantiles
+    # Compute the row medians of absolute differences. This means that we
+    # calculate how much the samples abundance accumulation differs from average
+    # abundance profile / accumulation in average.
+    difference <- rowMedians(abs(difference))
+    # Compute the relative change in absolute differences between the reference
+    # profile and each sample's quantiles. Identify the first index where this
+    # relative change exceeds the specified threshold. This index represents
+    # the count value at which the deviation between sample abundance levels and
+    # the reference profile becomes significant, highlighting a notable
+    # difference in abundance between samples
+    res <- abs(diff(difference)) / difference[-1]
+    res <- which(res > threshold)
+    res <- res[[1]]
+    # Convert the count value to a relative proportion by dividing by the
+    # maximum number of features. This scales the count to a proportion of the
+    # total feature count, normalizing it against the library size.
+    res <- res/max(found_features)
+    # # If the calculated percentile is less than or equal to 0.50, use 0.50 as
+    # # the default value
+    # if( res < 0.50 ){
+    #     res <- 0.50
+    # }
```

Comment:
I did not find any information on why is this 0.5. @antagomir DO you have information on this? I think this is not transparent to hardcode the lower limit especially as this does not seem to be part of the algorithm